Content.Shared/_ES/Auditions/ProducerComponent.cs

```diff
@@ -0,0 +1,84 @@
+using Content.Shared.Random;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared._ES.Auditions;
+
+/// <summary>
+/// This is the cast component placed onto the producer entity.
+/// </summary>
+[RegisterComponent]
+public sealed partial class ProducerComponent : Component
```

Comment:
should at least make these all datafields and mark the RelationshipContext as a DataDefinition. No reason not to support persistence for something so simple.